File: samples/csharp/src/Lists.cs
using System;
using System.Collections.Generic;

// UI settings
// Expression: arrayList
//   Type: System.Collections.Generic.List<int>
//     Layout: Array
//     Save: Object Attributes

// Expression: this.listOfLists
//   Type: System.Collections.Generic.List<System.Collections.Generic.List<int>>
//     Layout: Array2D
//     Save: Object Attributes
internal class Lists
{
    List<List<int>> listOfLists;
    public static void RunMain(string[] args)
    {
        new Lists().Start();
    }

    public void Start()
    {
        // list
        List<int> arrayList = new List<int>();
        arrayList.Add(1);
        arrayList.Add(2);

        // list of lists
        listOfLists = new List<List<int>>();
        List<int> a1 = new List<int>();
        a1.Add(1);
        a1.Add(2);
        List<int> a2 = new List<int>();
        a2.Add(3);
        a2.Add(4);
        listOfLists.Add(a1);
        listOfLists.Add(a2);

        Console.WriteLine("done");
    }
}

File: samples/ts/src/Lists.ts
// UI settings
// Expression: arrayList
//   Type: Array[]
//     Layout: Array
//     Save: Object Attributes

// Expression: this.listOfLists
//   Type: Array[][]
//     Layout: Array2D
//     Save: Object Attributes 
class Lists {
    listOfLists: number[][] | undefined;

    public static runMain() {
        new Lists().start();
    }

    public start() {
        // list
        const arrayList: number[] = [];
        arrayList.push(1);
        arrayList.push(2);

        // list of lists
        this.listOfLists = [];
        const a1: number[] = [];
        this.listOfLists.push(a1);
        a1.push(1);
        a1.push(2);
        
        const a2: number[] = [];
        this.listOfLists.push(a2);
        a2.push(3);
        a2.push(4);
        
        console.log("done");
    }
}

Lists.runMain();